util/eventbus/eventbustest/eventbustest.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package eventbustest
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"tailscale.com/util/eventbus"
+)
+
+func NewTestWatcher(d *eventbus.Debugger) *TestWatcher {
+	tw := &TestWatcher{
+		mon:     d.WatchBus(),
+		TimeOut: 5 * time.Second,
+		done:    make(chan bool, 1),
+		events:  make(chan any, 100),
+	}
+	go tw.watch()
+	return tw
+}
+
+type TestWatcher struct {
+	mon     *eventbus.Subscriber[eventbus.RoutedEvent]
+	events  chan any
+	done    chan bool
+	TimeOut time.Duration
+}
+
+// Expect is a particular implementation of ExpectFunc that tests for the
+// existence of an Event without caring about the contents of it.
+//
+// Example usage:
+//
+//	bus := eventbus.New()
+//	defer bus.Close()
+//
+//	tw := bus.Debugger().NewTestWatcher()
+//	defer tw.Done()
+//
+//	somethingThatEmitsSomeEvent()
+//	if err := eventbustest.Expect[EventFoo](tw); err != nil {
+//	  t.Error(err.Error())
+//	}
+func Expect[T any](tw *TestWatcher) error {
```

Comment:
These are fine as written, but another option to consider would be having these take a `testing.TB` as their first argument, so they can generate their own test errors, e.g.,

```go
func Expect[T any](t testing.TB, w *Watcher) {
   t.Helper()
   ExpectFunc(t, func(e T) bool { return true })
}

func ExpectFunc[T any](t testing.TB, w *Watcher, test func(T) bool) {
   t.Helper()
   
   // ...
   var zero T
   t.Errorf("timed out waiting for %T, saw %d events", zero, eventCount)
}
```

Not sure, though. The downside is that makes it harder to write runnable examples too, so maybe reporting the errors is better. What do you think?